flink-yarn/src/main/java/org/apache/flink/yarn/YarnClusterDescriptor.java

```diff
@@ -1781,6 +1781,12 @@ private void setClusterEntrypointInfoToConfig(final ApplicationReport report) {
 
         flinkConfiguration.set(
                 YarnConfigOptions.APPLICATION_ID, ConverterUtils.toString(clusterId));
+
+        // set cluster-id to app id if not specified
+        if (!flinkConfiguration.contains(HighAvailabilityOptions.HA_CLUSTER_ID)) {
```

Comment:
Should we deduplicate/sync this block with the one in `#startAppMaster`?

Reply:
Ok, let me see what I can simplify here.

Reply:
Ok, I've removed `YarnClusterDescriptor.zookeeperNamespace` as it seems to be no longer used.